### DISCUSSION TITLE
Feedback on v0.2.2 & PI Meeting

### DISCUSSION THREAD

**Discussion by noahmclean** [CIRDLES/Tripoli#137] [Ideas]
Feedback on existing alpha-version Tripoli v0.2.1, v0.2.2, and also other topics of discussion from June 2023 PI meeting.

**Comment by noahmclean** [comment]
Ensembles tab.
- [x] X-axis needs better axis labels -- higher resolution, at least 2-5 per axis length.  Y-axis needs fewer labels -- we don't really care about this absolute number, just the relative shape of the histogram.  
- [x] Stairstep histogram instead of individual bars. No white lines between bars.  Or at least, an option for stairstep vs. bars
- [x] Bigger font for the mean and standard deviation.  This is the most important information.
- [x] Vertically align the decimals in the mean and standard deviation.
- [x] When panning by mouse, stick the y minimum at zero (no physical meaning for negative counts)
- [ ] 2D histogram of the interpolated intensities (code provided by Scott) instead of knot values connected by lines for the intensities.

**Reply by bowring** [reply]
Need help on 2D histogram - make it an issue and provide more details, thanks

**Comment by noahmclean** [comment]
Convergences tab.
- [x] X-axis doesn't need any places after the decimal -- iterations are integer-valued.
- [x] Y-axis frequently has repeated axis labels.  Need to find a better way to label axis ticks.
- [x] Add window selector to choose which iterations (starting at 0) to discard as burn-in.  Burn-in window synchronizes across all plots in this tab.
- [x] Larger axis labels, please.

**Comment by noahmclean** [comment]
- [ ] Multi-block selection.  Shift- or control-select block numbers to see all the plots for those blocks.  Add rows for additional blocks at the bottom of the existing plots.

**Comment by noahmclean** [comment]
- [ ] Add "integration time" and "sequence time" fields, parsed from .TIMSAM method file.  These are 
- "integration time" = the time it takes to measure a row of data in the data file.  The XML field in the TIMSAM file is called "Integration Period" and is given as a string, e.g. "ms100" where the letters are the units (here, milliseconds) and the numbers are the value (here, 100).  This information is used in calculating d0.dsig, the data uncertainty vector in MassSpecTHB.m, replacing the 0.1 that's hard-coded there.  This value can possibly change per sequence (BL and OP).
- "sequence time" is the total time spent measuring a sequence.  The XML field in the TIMSAM file is called "Integration Time" and is given as a number, e.g. "10.0" and the units are assumed to be seconds.  This information will be used in the case where we have an uninformative data file (e.g., .TIMSDP) and a user-defined method file.  We'll use this information to estimate the uncertainty in the baseline measurement even though the baseline data is not provided.

**Comment by noahmclean** [comment]
## Baseline frequency options for Isotopx Phoenix and other mass specs.  
There are several baseline measurement frequency choices.
1. (Current assumption, most common): A baseline every block, measured at the start of the block.  
2. A baseline every cycle, usually measured at the start of the cycle.
3. A baseline at the start of the analysis.  No further baselines at the start of blocks or cycles.
4. A baseline every Monday or other irregular interval.  No information about the baseline in the data file, but a separate baseline file exists somewhere with the results of the baseline.

**Reply by bowring** [reply]
We will need exemplar files for these scenarios.

**Comment by noahmclean** [comment]
## Multiple Baseline Measurements

It's possible to have multiple baseline sequences (e.g., BL1, BL2, etc).  The method file contains information about which baseline is associated with on-peak sequence.  If multiple baselines are measured, then average the associated baselines (see cross ref column in OP table) before subtraction.  More coming from Scott. Scott proposes moving baseline processing to the head of the process.

**Reply by bowring** [reply]
Will need exemplar files, thanks.

**Comment by noahmclean** [comment]
## Sequence and Baseline Table: Add Info
- [ ] Add a two columns to both the sequence and baseline tables.  The first column is the sequence time (see comment above), and the second column is the integration time (see column above).  These numbers exist for every sequence.
Note: we are not using the sequence time in our informative-file MCMC code.  It's just for display/info.
